Sources/Sentry/Public/SentrySDK.h

```diff
@@ -24,6 +24,12 @@ SENTRY_NO_INIT
  */
 @property (class, nonatomic, readonly) BOOL isEnabled;
 
+/**
+ * Inits and configures Sentry (SentryHub, SentryClient) and sets up all integrations. Make sure to
+ * set a valid DSN.
+ */
++ (void)startWithOptions:(SentryOptions *)options andScope:(nullable SentryScope *)scope NS_SWIFT_NAME(start(options:scope:));
```

Comment:
`h`: I prefer to align with JavaScript instead of adding an extra method. We should add a method `initialScope` to the options, basically what we have for `SentrySDK.configureScope`. The public-facing API should look like this
```Swift
SentrySDK.start { options in
    options.dsn = "dsn"
    options.initialScope { scope in
        scope.setEnvironment("debug")
        // ...
    }
}
```

I also think that the API above is cleaner than having to init the options and the scope yourself:
```
let options = Options()
options.dns = "dsn"

let initialScope = Scope()
scope.setEnvironment("debug")

SentrySDK.start(options: options, scope: initialScope)
```

Reply:
Gotcha, done! I also made it possible to pass a whole `Scope` object by setting `initialScopeFactory` directly instead of calling `initialScope:`, which should solve for use case 2 mentioned in the PR description.